my-vault/internal/handlers/handlers.go

```diff
@@ -5,19 +5,14 @@ import (
 	"errors"
 	"fmt"
 	"net/http"
-	"sync"
 
+	"github.com/bmasaiti/go-projects/my-vault/internal/domain"
+	"github.com/bmasaiti/go-projects/my-vault/internal/storage"
 	"github.com/google/uuid"
 )
 
-type Secret struct {
-	Id    string
-	Name  string
-	KVMap map[string]string
-}
 
-var mu sync.Mutex
-var db = make(map[string]Secret)
+var db = storage.NewInMemorySecretRepo()
```

Comment:
Instead of this, I think we should have a

```
type Handler struct {
    DB SecretsRepository
}
```

Then in `main.go`, we do

```
h := handlers.Handler{DB: storage.NewInMemorySecretRepo()}
```

This encapsulates the `handlers` package from the _implementation_ of the `SecretsRepository`. It will still need to use the `storage.Secret` type, but that's okay.